### PR TITLE
Feature/cs 030 6/no duplicate code analyzer algorithm

### PR DESCRIPTION
**Pull request by luiggy-mamani-jalafund**
Implemented LSH algorithm to detect duplicate code.

#### Resolved task: ([CS-030-6 [SP-3]: No Duplicate Code Analyzer](https://app.clickup.com/t/868ay9gr7))
* * * *

## What's in this pull request?
Implemented an algorithm to detect duplicate code using LSH which needs steps provided by other modules like normalization, shingles generation, min hash calculator, and the LSH candidate selector.

## Type of change
- [X] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?
Unit tests have been added for each module of the algorithm.

## Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] I have checked my code and corrected any misspellings

- - -
This change is [Reviewable](https://reviewable.io/reviews/SoftDevSix/Argos/51)

### CODE REVIEW

**Review by diego-centella-jalafund — APPROVED** [review]

File: ASTProcessor/src/test/java/edu/usb/argos/astprocessor/analyzer/codeSmells/AntlrMethodNormalizerTest.java [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+        Optional<JavaParser.MethodDeclarationContext> method = Optional.ofNullable(methods.get(0));
+        assertTrue(method.isPresent());
+
+        List<String> tokens = methodNormalizer.normalize(method.get());
+        int expectedTokensSize = 66;
+
+        assertEquals(expectedTokensSize, tokens.size());
+
+//      FOR, LPAREN, INT, IDENTIFIER, ASSIGN, LITERAL, SEMI,
```
Comment (salet-gutierrez-jalafund) [inline ASTProcessor/src/test/java/edu/usb/argos/astprocessor/analyzer/codeSmells/AntlrMethodNormalizerTest.java:94]: Why are these comments in the tests, could you delete it or is it necessary?

**Review by salet-gutierrez-jalafund — CHANGES_REQUESTED** [review]

File: ASTProcessor/src/test/java/edu/usb/argos/astprocessor/analyzer/codeSmells/AntlrMethodNormalizerTest.java (hunk 1 above)
Comment (luiggy-mamani-jalafund) [reply ASTProcessor/src/test/java/edu/usb/argos/astprocessor/analyzer/codeSmells/AntlrMethodNormalizerTest.java:94]: The test is a method normalizer using tokens to reduce noise during duplicate code analysis, I put those comments to show the tokens that are being generated based on a statement

Comment (salet-gutierrez-jalafund) [reply ASTProcessor/src/test/java/edu/usb/argos/astprocessor/analyzer/codeSmells/AntlrMethodNormalizerTest.java:94]: ok

**Review by salet-gutierrez-jalafund — APPROVED** [review]
Good Job

**Pull request merged**